javascript/javascript1/week2/homework4/weatherClothApp.js

```diff
@@ -0,0 +1,75 @@
+const winterClothes = {
+    coats: ['Winter Coat1', 'Winter Coat2',],
+    headwear: ['beanie', 'hat'],
+    scarves: ['Heavy Scarf1', 'Heavy Scarf 2'],
+    tops: ['turtle-neck', 'wool sweater', 'thick hoodie', 'crew neck sweater'],
+    gloves: ["Winter gloves"],
+    pants: ['Winter pants1', 'rain pants', 'Winter pants2'], 
+    socks: ['winter socks', 'chrismas socks', 'wool socks'],
+    shoes: ['winter boots', 'leather boots', 'high boots']
+}
+
+const fallSpringClothes = {
+    coats: ['light jacket1', 'light jacket2'],
+    scarves: ['light scarf1', 'light scarf2'],
+    tops: ['warm top', 'thick shirt', 'sweater', 'hoodie'],
+    gloves: ['light gloves'],
+    pants: ['grey pants', 'rain pants', 'blue pants'],
+    socks: ['regular socks'],
+    shoes: ['leather boots', 'light shoes', 'sport shoes', 'sneackers']
+}
+
+const summerClothes = {
+    
+    tops: ['light shirt', 't-shirt1', 't-shirt2', 'v-neck', 'light shirt2'],
+    headwear: ['baseball cap', 'bob'],
+    pants: ['light pants', 'shorts'],
+    socks: ['regular socks'], 
+    shoes: ['fliflops', 'sandals', 'light sneackers']
+}
+
+function getRandomValues(obj) {
+    const result = {};
+    for (let key in obj) {
+        const valuesArray = obj[key];
+        const randomIndex = Math.floor(Math.random() * valuesArray.length);
+        result[key] = valuesArray[randomIndex];
+    }
+    return result;
+}
+
+function getClothes(degrees){
+    if (degrees === undefined) { return `please enter and int value of the temperature in Celsius`}
```

Comment:
I have never seen anyone go into this amount of detail on this task, and I love it! 🤩 Excellent use of template strings as well. 😄 

As there are some complex features and tricks in use, I just wanna hear you explain some of the concepts in your own word:
 - What does a `for..in` loop do? And how does it relate to arrays?
 - What does `clothes.coats || 'None'` do and how does it work?

Should you get bored, I think this function would benefit greatly from another helper function that takes a `clothes` object and pretty-prints based on the properties. In that way, you could shorten it a lot! **I do not expect you to do this**, but I think it would be a good exercise to push your skills further if you want to. 😄

Reply:
This was really instructive. I am really grateful for the very instructive and illuminating advice. I have added the function you suggested, and it did make this a lot more readable, and that was not because I was bored in any way 😄😄
To answer your question: 

- The `for..in` iterates through a certain interval that we determine, while performing a task. in an array this specific syntax will iterate through the indices of the said array.  However I have just read through some documentation and got reminded that: `for..in` does not only iterate through the content of the array, but also through properties that are not necessarily indexed or enumerated. So if I understand correctly, I should better use `for...of` instead, right? 
- `clothes.coats || 'None'` checks for left value first, and if it's falsy it will check the one on the right. I used this as I was still populating the objects, and I didn't want an `undefined` value showing up. As things stand, I don't need it, especially since now there's a function to "pretty-print" that part. 

This was a lot of fun, I am very grateful.

Reply:
Regarding `for...in` and `for...of`, put very simply the former loops through properties and the latter through elements. If you do `in` on an array, it spits out the indexes (and any properties that it might have as well) which is a bit error-prone. If you need to iterate through arrays, `of` is definitely the safer choice as it just yields the values. 😄 

It also sounds like you kinda encountered a gotcha with `in`: that it can pick up on weird, unexpected properties. You might have seen `hasOwnProperty` already which is generally how you'd avoid this. But yea, we are getting way ahead of ourselves here. 😆 

Good works and thanks for humoring me! 🤩

Reply:
Best deal ever, I get knowledge, you get laughs 😄 
I am very grateful. This was really insightful.